### DISCUSSION TITLE
"installation not allowed to Write organization package" from forks

### DISCUSSION THREAD

**Discussion by awoimbee** [community/community#69331] [Packages]
### Select Topic Area

Question

### Body

Hi,
I have [a project](https://github.com/Trow-Registry/trow) where I would need github actions, running on PRs from forks, to push to the container registry.
I have [a job that builds an image](https://github.com/Trow-Registry/trow/blob/70d580ad1fe69ce36782b0540a47ccf6b464a477/.github/workflows/pr-tests.yaml#L34) and then [another job that uses the image as a service](https://github.com/Trow-Registry/trow/blob/70d580ad1fe69ce36782b0540a47ccf6b464a477/.github/workflows/pr-tests.yaml#L76), so I absolutely need to push the image to a registry, which currently is https://github.com/Trow-Registry/trow/pkgs/container/trow-dev.

My issue is that PRs from forks get an error message "`installation not allowed to Write organization package`" ([see](https://github.com/Trow-Registry/trow/actions/runs/6318633863/job/17270544484)).
Is there a way to bypass this error ?

Kinda relates to https://github.com/orgs/community/discussions/57724

**Comment by Dipanshu089** [comment]
message you're encountering,  typically occurs when GitHub Actions in a forked repository attempt to publish a container image to a package registry owned by the base repository or organization. By default, GitHub does not allow forks to publish to certain sensitive destinations to prevent unauthorized access.

**To allow GitHub Actions from forks to push container images to your container registry, follow these simple steps:**
- Create a Personal Access Token (PAT) with image push permissions.
- Add the PAT as a secret in your base repository.
- Modify your GitHub Actions workflow to use this secret for Docker authentication when pushing images.
- Ask contributors to add the same PAT secret to their forked repositories.

This way, GitHub Actions in forked repositories can authenticate and push images to your registry without encountering permission errors.

**Reply by afbase** [reply]
This is the information I've been looking for! thank you for this answer!

**Comment by awoimbee** [comment]
Ok nevermind, I followed https://docs.docker.com/build/ci/github-actions/share-image-jobs/ and it works. It adds more steps to my jobs and I can't use the `services` thing, but it seems to work.
(see https://github.com/Trow-Registry/trow/pull/358)
